{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/xgboost/core.py:265: FutureWarning: Your system has an old version of glibc (< 2.28). We will stop supporting Linux distros with glibc older than 2.28 after **May 31, 2025**. Please upgrade to a recent Linux distro (with glibc 2.28+) to use future versions of XGBoost.\n",
      "Note: You have installed the 'manylinux2014' variant of XGBoost. Certain features such as GPU algorithms or federated learning are not available. To use these features, please upgrade to a recent Linux distro with glibc 2.28+, and install the 'manylinux_2_28' variant.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "import MLMarker_app\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from MLMarker_app import MLMarker\n",
    "import matplotlib.pyplot as plt\n",
    "import random\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4120 model features are not in the sample and were added as zero values\n",
      "Warning: 350 sample features are not in the model and were removed\n"
     ]
    }
   ],
   "source": [
    "proteins = [\"P68871\",\"P02042\",\"P69891\",\"P69892\",\"P02100\",\"P69905\",\"P02008\",\"P02768\",\"P04264\",\"P13645\",\"P07477\",\"P00915\",\"P01023\",\"P01009\",\"P02787\",\"P01024\",\"Q6PGP7\",\"P32119\",\"Q06830\",\"P02647\",\"Q13228\",\"P04040\",\"P30043\",\"P00918\",\"P01834\",\"P00738\",\"P02766\",\"P02790\",\"P01876\",\"P04406\",\"P01857\",\"P01859\",\"P01861\",\"P01011\",\"P60709\",\"P00450\",\"P30041\",\"P60174\",\"P81605\",\"P04217\",\"P01008\",\"P10599\",\"P02763\",\"P19652\",\"P00491\",\"P62979\",\"P11142\",\"P19823\",\"Q14624\",\"P00441\",\"P0DOY2\",\"B9A064\",\"P07195\",\"P00338\",\"P25311\",\"P05155\",\"P02765\",\"P06733\",\"P09104\",\"P01040\",\"P37840\",\"P06727\",\"P19827\",\"P43652\",\"P00352\",\"P05091\",\"P06702\",\"P37837\",\"P10909\",\"A0A0C4DH25\",\"Q7Z6K1\",\"P07738\",\"P08185\",\"Q99497\",\"P78417\",\"P0C0L5\",\"P00751\",\"P02652\",\"P62937\",\"P13716\",\"P11171\",\"P01042\",\"P02656\",\"Q86YZ3\",\"P02774\",\"P30086\",\"O75368\",\"P01019\",\"Q96JH7\",\"P00558\",\"P23526\",\"P15531\",\"P22392\",\"P16157\",\"P0DP23\",\"A0A0A0MRZ7\",\"P07451\",\"P04075\",\"P30046\",\"Q02413\",\"P23528\",\"P02760\",\"P02671\",\"Q7KZ85\",\"P29508\",\"P51884\",\"P20810\",\"P02753\",\"P00568\",\"P02743\",\"P01591\",\"Q8TF72\",\"Q9NZD4\",\"P06703\",\"P22061\",\"A0A0B4J1X5\",\"P02730\",\"Q13404\",\"P54725\",\"P01700\",\"O14618\",\"P15924\",\"O75223\",\"P26447\",\"P00492\",\"P31151\",\"P06396\",\"P05109\",\"P40925\",\"P01871\",\"P01780\",\"P02750\",\"B5ME19\",\"P05543\",\"P48637\",\"P02749\",\"Q5D862\",\"P09211\",\"P05546\",\"Q9H3R0\",\"Q9H299\",\"A0A075B6K5\",\"Q04760\",\"Q9BS40\",\"P31025\",\"P04196\",\"Q9NZT1\",\"P05090\",\"P62942\",\"P49247\",\"P31944\",\"A0A0A0MRZ8\",\"Q9UNZ2\",\"P61626\",\"O00233\",\"Q06323\",\"A0A075B6R9\",\"P06744\",\"P0DP06\",\"A0A075B6H7\",\"P25325\",\"P20930\",\"Q9NZP8\",\"P62826\",\"Q96PD5\",\"Q9NY33\",\"Q9GZP4\",\"Q13867\",\"P06312\",\"P04004\",\"Q86U44\",\"Q8IUG5\",\"A7E2Y1\",\"Q9ULV0\",\"P61088\",\"P17174\",\"P35612\",\"P24666\",\"Q5T750\",\"O75882\",\"Q9H9B4\",\"P13489\",\"P16152\",\"Q8TC76\",\"P18669\",\"P63208\",\"Q16775\",\"Q9H0T7\",\"Q6SA08\",\"P05156\",\"A0A0B4J1V0\",\"Q92974\",\"P35754\",\"Q9NPA2\",\"P61956\",\"P02724\",\"Q86T24\",\"P61970\",\"P10768\",\"P09105\",\"Q05D60\",\"P21333\",\"P02452\",\"P26639\",\"P14923\",\"P01602\",\"Q5TDH0\",\"Q15149\",\"A0A075B6I0\",\"P43487\",\"P28066\",\"P49773\",\"P06681\",\"P01593\",\"Q5U651\",\"Q3YEC7\",\"O75154\",\"Q9HC38\",\"Q99698\",\"P52566\",\"P08697\",\"P62258\",\"P68402\",\"P12955\",\"Q6XQN6\",\"Q969T9\",\"P14174\",\"Q08380\",\"A0A0C4DH34\",\"P07339\",\"Q01469\",\"P08123\",\"P07357\",\"P05089\",\"Q6P2E9\",\"Q96IU4\",\"Q9H2D6\",\"O75896\",\"P04211\",\"Q93100\",\"P58546\",\"O60513\",\"Q96DR8\",\"A6NGG8\",\"Q9Y4W6\",\"Q13630\",\"A0A0C4DH38\",\"O95747\",\"P05452\",\"Q08554\",\"Q9NRV9\",\"P18206\",\"P16949\",\"Q9ULT8\",\"P07741\",\"Q9BWD1\",\"Q9UHP6\",\"Q6KC79\",\"O15027\",\"P08603\",\"Q03591\",\"Q9UGM3\",\"Q86YF9\",\"Q9NQG5\",\"Q8IWJ2\",\"O60522\",\"P02549\",\"Q08378\",\"O75970\",\"P07203\",\"Q86VP3\",\"Q5HYW2\",\"P08397\",\"O75618\",\"Q9NR99\",\"Q96PN6\",\"O75151\",\"Q9P1Z9\",\"O43598\",\"O94919\",\"Q3SY00\",\"Q9Y2E4\",\"Q96F24\",\"Q9Y592\",\"Q6ZN84\",\"Q6ZMN7\",\"Q9H6N6\",\"Q96T58\",\"Q96KN7\",\"Q92834\",\"P01715\",\"Q9BZE3\",\"Q16610\",\"P62834\",\"P58004\",\"P07355\",\"P03952\",\"P63241\",\"Q9P2N6\",\"Q8NDH2\",\"Q15772\",\"O14733\",\"Q96D46\",\"Q6NX49\",\"P27469\",\"Q7Z7H5\",\"Q13409\",\"P50747\",\"Q8N3Y1\",\"P12273\",\"P27986\",\"P00748\",\"P28072\",\"P55157\",\"O94818\",\"Q9UJX6\",\"P02655\",\"Q6ZQQ6\",\"P50395\",\"Q86SQ4\",\"Q9BXL6\",\"Q9HBK9\",\"P23443\",\"Q8TCW9\",\"Q8IYE0\",\"Q9H4B6\",\"Q32M84\",\"Q8N5P1\",\"P00167\",\"Q9NX36\",\"P10643\",\"Q9ULL4\",\"Q14643\",\"P02775\",\"O95859\",\"Q8IYL9\",\"Q8NCX0\",\"Q96RW7\",\"P17900\",\"Q5VTE6\",\"Q9Y2K3\",\"Q9Y2Q9\",\"O60939\",\"Q15822\",\"Q12791\",\"O60503\",\"Q5KU26\",\"Q6PII3\",\"P48506\",\"P09769\",\"Q8IYA7\",\"O75354\",\"Q12912\",\"P48664\",\"Q8WY21\",\"Q9Y6I3\",\"Q8IZH2\",\"Q08188\",\"P08253\",\"Q15256\",\"Q14789\",\"Q86SF2\",\"Q5VZ66\",\"Q01726\",\"P08684\",\"P26717\",\"P55916\",\"Q14146\",\"Q13201\",\"P49189\",\"P08217\",\"Q96T23\",\"P52565\",\"Q5T749\",\"Q6UXT8\",\"Q15700\",\"Q6ZP01\",\"O75347\",\"Q9P2E3\",\"Q6UWP2\",\"Q9C0H9\",\"A8MT19\",\"Q9H7P6\",\"O95390\",\"Q03001\",\"Q9HAR2\",\"Q9NP72\",\"Q9H9Y2\",\"Q96G03\",\"O95759\",\"Q96SB4\",\"P29317\",\"A0A8I5KY20\",\"Q9BY77\",\"P11766\",\"Q86UK0\",\"O43395\",\"Q96LB3\",\"Q5U5Z8\",\"O94911\",\"Q7Z3V4\",\"O15049\",\"Q08999\",\"Q96EY7\",\"Q14CX7\",\"Q99569\",\"P18433\",\"Q6DKK2\",\"Q14671\",\"Q5T7N2\",\"Q9NPA8\",\"P50616\",\"O60427\",\"Q96QV6\",\"Q9BZ19\",\"A8K7I4\",\"Q2VPK5\",\"Q96C34\",\"Q9NR56\",\"Q9BQ65\",\"P82094\",\"O95831\",\"Q96JN0\",\"O15020\",\"Q86YM7\",\"Q9BX95\",\"P45985\",\"Q9Y696\",\"Q9Y4D1\",\"P25054\",\"Q8NI27\",\"P29074\",\"Q7L5Y6\",\"A6NE02\",\"Q99536\",\"Q93034\",\"Q8N1I0\",\"Q14676\",\"P55196\",\"Q96JB1\",\"Q14141\",\"Q502W6\",\"P05538\",\"Q9H3E2\",\"Q8IYT8\",\"Q8WXG6\",\"Q9C0I3\",\"Q6AHZ1\",\"Q7Z7L1\",\"O15350\",\"P09417\",\"Q8N554\",\"O43396\",\"Q15075\",\"Q9UMS6\",\"Q2M2H8\",\"O14578\",\"P0C7I6\",\"P37287\",\"Q9HA64\",\"O60678\",\"Q8IUR6\",\"Q8N119\",\"O75122\",\"Q9UDY2\",\"P36406\",\"Q9Y3B3\",\"Q96SN8\",\"Q02224\",\"Q9NQW1\",\"P52272\",\"Q9P0T4\",\"Q96P44\",\"Q7LBC6\",\"Q4V328\",\"Q9UMF0\",\"Q9NY35\",\"Q9UKZ4\",\"Q6UW63\",\"Q00613\",\"Q09666\",\"Q4FZB7\",\"Q96F07\",\"Q6UXY8\",\"Q9UJA5\",\"Q9Y5B9\",\"Q9NW08\",\"Q9UL46\",\"P08581\",\"Q6P1X5\",\"Q9HBH7\",\"O15119\",\"Q96DR5\",\"O14681\",\"Q96MT7\",\"Q9NY57\",\"P0C7V7\",\"O75694\",\"A1A5B4\",\"P08575\",\"Q8NCJ5\",\"Q96F44\",\"Q6K0P9\",\"P56179\",\"P36955\",\"Q9Y4G2\",\"O94806\",\"P52735\",\"Q9NZN5\",\"O60783\",\"Q6V1P9\",\"Q9UQ84\",\"P23083\",\"Q709C8\",\"Q9UPN3\",\"Q9UK61\",\"Q9H361\",\"O15294\",\"Q0VDD8\",\"P38606\",\"Q8ND04\",\"A6NNT2\",\"Q92530\",\"O95183\",\"Q5VU65\",\"Q9NYU2\",\"Q7Z2Y8\",\"A6H8Y1\",\"Q8TC84\",\"O95171\",\"Q12802\",\"Q14435\",\"P56556\",\"Q9C0D5\",\"Q9NSC2\",\"Q8TC99\",\"Q9UBR2\",\"Q03181\",\"Q86VE3\",\"O95625\",\"Q9H079\",\"Q6WCQ1\",\"Q8N7W2\",\"Q92614\",\"Q9Y2Q0\",\"A4D1F6\",\"P62195\",\"Q86X59\",\"Q9NP98\",\"M0R2J8\",\"Q9H0U9\",\"P59044\",\"Q6ZR52\",\"Q7Z465\",\"Q9Y5E8\",\"C9JXX5\",\"Q6P4F7\",\"Q8NDL9\",\"Q9NVF7\",\"O60673\",\"Q5TAQ9\",\"Q8WZ42\",\"Q96PE6\",\"Q6P4E1\",\"Q9H2G9\",\"Q6V0I7\",\"Q96CW1\",\"Q12887\",\"Q9H8M5\",\"Q8IXQ8\",\"Q9NT68\",\"Q9NTJ3\",\"Q16666\",\"Q16798\",\"Q9ULL8\",\"P30626\",\"Q99963\",\"Q8N4C6\",\"P62820\",\"Q9Y259\",\"Q92585\",\"Q4L180\",\"Q9P2E9\",\"Q6UXB4\",\"Q8NFY4\",\"Q9H9Y6\",\"Q9UKC9\",\"O75764\",\"Q86VS3\",\"Q5VUJ6\",\"Q9UHC1\",\"P36543\",\"Q8TBY9\",\"Q8N5C6\",\"P20929\",\"O14523\",\"Q9HB63\",\"P36888\",\"P09960\",\"Q9Y6C5\",\"Q96C45\",\"Q68DK2\",\"Q99726\",\"O94898\",\"Q8WWN8\",\"P49137\",\"Q9BV73\",\"O75132\",\"P16144\",\"A2RRP1\",\"O75554\",\"Q6ZS02\",\"Q96PV4\",\"O00192\"]\n",
    "data = np.ones((1, len(proteins)))\n",
    "predict_df = pd.DataFrame(data, columns=proteins)\n",
    "predict_df.head()\n",
    "test = MLMarker(predict_df.iloc[0:1,:], binary = False, dev=False)\n",
    "prediction = MLMarker.adjusted_absent_shap_values_df(test, penalty_factor=1)\n",
    "prediction.to_csv('/home/compomics/git/MLMarker/Projects/Laura/Messen_predictie.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "features_path = \"/home/compomics/git/MLMarker/models/features_TP_full_92%_10exp_2024.txt\"\n",
    "with open(features_path, 'r') as features_file:\n",
    "    MLM_features = features_file.read().split(',\\n')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "prediction[list(set(proteins).intersection(MLM_features))].to_csv('Messen_predictie_subset_present.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>P02775</th>\n",
       "      <th>P10599</th>\n",
       "      <th>P07357</th>\n",
       "      <th>P18206</th>\n",
       "      <th>P02100</th>\n",
       "      <th>Q14CX7</th>\n",
       "      <th>O94919</th>\n",
       "      <th>P00748</th>\n",
       "      <th>Q96G03</th>\n",
       "      <th>Q9Y2K3</th>\n",
       "      <th>...</th>\n",
       "      <th>Q13409</th>\n",
       "      <th>Q5U651</th>\n",
       "      <th>O75882</th>\n",
       "      <th>P05091</th>\n",
       "      <th>P07451</th>\n",
       "      <th>Q13630</th>\n",
       "      <th>Q9Y5B9</th>\n",
       "      <th>P36955</th>\n",
       "      <th>P35612</th>\n",
       "      <th>P25311</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Monocytes</th>\n",
       "      <td>0.000838</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>-0.000072</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000010</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000586</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-6.795752e-07</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.001593</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000010</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Skeletal muscle</th>\n",
       "      <td>-0.000008</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-0.000001</td>\n",
       "      <td>-0.000018</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007836</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.085237e-05</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>0.013506</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Salivary gland</th>\n",
       "      <td>-0.000216</td>\n",
       "      <td>-2.724310e-06</td>\n",
       "      <td>-0.000019</td>\n",
       "      <td>-0.001244</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000014</td>\n",
       "      <td>-0.000016</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>-0.000697</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-3.479472e-04</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>-0.000004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-7.657845e-07</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ovary</th>\n",
       "      <td>-0.000436</td>\n",
       "      <td>6.149082e-07</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>-0.000332</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>-0.000012</td>\n",
       "      <td>0.000216</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>-0.000533</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-5.596159e-06</td>\n",
       "      <td>-4.597069e-04</td>\n",
       "      <td>0.000059</td>\n",
       "      <td>-0.000745</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>-5.596117e-07</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>-0.000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B-cells</th>\n",
       "      <td>0.001339</td>\n",
       "      <td>-8.399098e-05</td>\n",
       "      <td>-0.000037</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000021</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000011</td>\n",
       "      <td>-0.000341</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-5.515997e-07</td>\n",
       "      <td>-9.329886e-05</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>-0.000425</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000004</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000003</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 264 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   P02775        P10599    P07357    P18206    P02100  \\\n",
       "tissue                                                                  \n",
       "Monocytes        0.000838  0.000000e+00  0.000001 -0.000072  0.000000   \n",
       "Skeletal muscle -0.000008  0.000000e+00 -0.000001 -0.000018  0.000000   \n",
       "Salivary gland  -0.000216 -2.724310e-06 -0.000019 -0.001244  0.000007   \n",
       "Ovary           -0.000436  6.149082e-07  0.000036 -0.000332  0.000035   \n",
       "B-cells          0.001339 -8.399098e-05 -0.000037  0.000025  0.000000   \n",
       "\n",
       "                   Q14CX7    O94919    P00748    Q96G03    Q9Y2K3  ...  \\\n",
       "tissue                                                             ...   \n",
       "Monocytes        0.000000 -0.000010  0.000000  0.000000 -0.000586  ...   \n",
       "Skeletal muscle  0.000000  0.000033  0.000000  0.000000  0.007836  ...   \n",
       "Salivary gland   0.000000 -0.000014 -0.000016  0.000028 -0.000697  ...   \n",
       "Ovary           -0.000012  0.000216  0.000032  0.000037 -0.000533  ...   \n",
       "B-cells          0.000000 -0.000021  0.000000 -0.000011 -0.000341  ...   \n",
       "\n",
       "                 Q13409        Q5U651        O75882    P05091    P07451  \\\n",
       "tissue                                                                    \n",
       "Monocytes           0.0  0.000000e+00 -6.795752e-07  0.000000 -0.001593   \n",
       "Skeletal muscle     0.0  0.000000e+00 -1.085237e-05  0.000002  0.013506   \n",
       "Salivary gland      0.0  0.000000e+00 -3.479472e-04  0.000033 -0.000004   \n",
       "Ovary               0.0 -5.596159e-06 -4.597069e-04  0.000059 -0.000745   \n",
       "B-cells             0.0 -5.515997e-07 -9.329886e-05  0.000001 -0.000425   \n",
       "\n",
       "                   Q13630    Q9Y5B9        P36955    P35612    P25311  \n",
       "tissue                                                                 \n",
       "Monocytes        0.000000 -0.000010  0.000000e+00  0.000000  0.000000  \n",
       "Skeletal muscle  0.000000  0.000000  0.000000e+00  0.000000 -0.000021  \n",
       "Salivary gland   0.000000  0.000000 -7.657845e-07  0.000000  0.000016  \n",
       "Ovary            0.000008  0.000008 -5.596117e-07  0.000002 -0.000023  \n",
       "B-cells          0.000000 -0.000004  0.000000e+00  0.000000 -0.000003  \n",
       "\n",
       "[5 rows x 264 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_proteins = prediction[list(set(proteins).intersection(MLM_features))]\n",
    "subset_proteins = subset_proteins.iloc[:5, :]\n",
    "subset_proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 Index(['O94919', 'Q9Y2K3', 'P00352', 'P02787', 'P51884', 'Q93100', 'P01023',\n",
      "       'P02452', 'Q9NP98', 'P68871', 'P02549', 'P17174', 'Q15772', 'P05091',\n",
      "       'P07451'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "muscle_proteins = subset_proteins.loc[:, subset_proteins.loc['Skeletal muscle'] > 0].columns\n",
    "print(len(muscle_proteins), muscle_proteins)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0070062</td>\n",
       "      <td>extracellular exosome</td>\n",
       "      <td>1.469401e-58</td>\n",
       "      <td>True</td>\n",
       "      <td>\"A vesicle that is released into the extracell...</td>\n",
       "      <td>2109</td>\n",
       "      <td>230</td>\n",
       "      <td>120</td>\n",
       "      <td>22097</td>\n",
       "      <td>0.521739</td>\n",
       "      <td>0.056899</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0005615, GO:1903561]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:1903561</td>\n",
       "      <td>extracellular vesicle</td>\n",
       "      <td>5.212769e-58</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Any vesicle that is part of the extracellular...</td>\n",
       "      <td>2133</td>\n",
       "      <td>230</td>\n",
       "      <td>120</td>\n",
       "      <td>22097</td>\n",
       "      <td>0.521739</td>\n",
       "      <td>0.056259</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0031982, GO:0065010]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0043230</td>\n",
       "      <td>extracellular organelle</td>\n",
       "      <td>5.493232e-58</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Organized structure of distinctive morphology...</td>\n",
       "      <td>2134</td>\n",
       "      <td>230</td>\n",
       "      <td>120</td>\n",
       "      <td>22097</td>\n",
       "      <td>0.521739</td>\n",
       "      <td>0.056232</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0005576, GO:0043226]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0065010</td>\n",
       "      <td>extracellular membrane-bounded organelle</td>\n",
       "      <td>5.493232e-58</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Organized structure of distinctive morphology...</td>\n",
       "      <td>2134</td>\n",
       "      <td>230</td>\n",
       "      <td>120</td>\n",
       "      <td>22097</td>\n",
       "      <td>0.521739</td>\n",
       "      <td>0.056232</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0043227, GO:0043230]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GO:CC</td>\n",
       "      <td>GO:0005615</td>\n",
       "      <td>extracellular space</td>\n",
       "      <td>1.299284e-49</td>\n",
       "      <td>True</td>\n",
       "      <td>\"That part of a multicellular organism outside...</td>\n",
       "      <td>3303</td>\n",
       "      <td>230</td>\n",
       "      <td>134</td>\n",
       "      <td>22097</td>\n",
       "      <td>0.582609</td>\n",
       "      <td>0.040569</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0005576, GO:0110165]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>HPA</td>\n",
       "      <td>HPA:0470183</td>\n",
       "      <td>skin 2; cells in granular layer[High]</td>\n",
       "      <td>4.624372e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>skin 2; cells in granular layer[High]</td>\n",
       "      <td>248</td>\n",
       "      <td>176</td>\n",
       "      <td>13</td>\n",
       "      <td>11019</td>\n",
       "      <td>0.073864</td>\n",
       "      <td>0.052419</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HPA:0470182]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>HPA</td>\n",
       "      <td>HPA:0471413</td>\n",
       "      <td>skin 2; cells in spinous layer[High]</td>\n",
       "      <td>4.624372e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>skin 2; cells in spinous layer[High]</td>\n",
       "      <td>248</td>\n",
       "      <td>176</td>\n",
       "      <td>13</td>\n",
       "      <td>11019</td>\n",
       "      <td>0.073864</td>\n",
       "      <td>0.052419</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[HPA:0471412]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>GO:BP</td>\n",
       "      <td>GO:0002252</td>\n",
       "      <td>immune effector process</td>\n",
       "      <td>4.822972e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Any process of the immune system that execute...</td>\n",
       "      <td>698</td>\n",
       "      <td>227</td>\n",
       "      <td>21</td>\n",
       "      <td>21031</td>\n",
       "      <td>0.092511</td>\n",
       "      <td>0.030086</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0002376]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>187</th>\n",
       "      <td>GO:MF</td>\n",
       "      <td>GO:0019145</td>\n",
       "      <td>aminobutyraldehyde dehydrogenase activity</td>\n",
       "      <td>4.966734e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Catalysis of the reaction: 4-aminobutanal + N...</td>\n",
       "      <td>2</td>\n",
       "      <td>223</td>\n",
       "      <td>2</td>\n",
       "      <td>20212</td>\n",
       "      <td>0.008969</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0004029]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>GO:MF</td>\n",
       "      <td>GO:0008545</td>\n",
       "      <td>JUN kinase kinase activity</td>\n",
       "      <td>4.966734e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>\"Catalysis of the phosphorylation of tyrosine ...</td>\n",
       "      <td>2</td>\n",
       "      <td>223</td>\n",
       "      <td>2</td>\n",
       "      <td>20212</td>\n",
       "      <td>0.008969</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[GO:0004708]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>189 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    source       native                                       name  \\\n",
       "0    GO:CC   GO:0070062                      extracellular exosome   \n",
       "1    GO:CC   GO:1903561                      extracellular vesicle   \n",
       "2    GO:CC   GO:0043230                    extracellular organelle   \n",
       "3    GO:CC   GO:0065010   extracellular membrane-bounded organelle   \n",
       "4    GO:CC   GO:0005615                        extracellular space   \n",
       "..     ...          ...                                        ...   \n",
       "184    HPA  HPA:0470183      skin 2; cells in granular layer[High]   \n",
       "185    HPA  HPA:0471413       skin 2; cells in spinous layer[High]   \n",
       "186  GO:BP   GO:0002252                    immune effector process   \n",
       "187  GO:MF   GO:0019145  aminobutyraldehyde dehydrogenase activity   \n",
       "188  GO:MF   GO:0008545                 JUN kinase kinase activity   \n",
       "\n",
       "          p_value  significant  \\\n",
       "0    1.469401e-58         True   \n",
       "1    5.212769e-58         True   \n",
       "2    5.493232e-58         True   \n",
       "3    5.493232e-58         True   \n",
       "4    1.299284e-49         True   \n",
       "..            ...          ...   \n",
       "184  4.624372e-02         True   \n",
       "185  4.624372e-02         True   \n",
       "186  4.822972e-02         True   \n",
       "187  4.966734e-02         True   \n",
       "188  4.966734e-02         True   \n",
       "\n",
       "                                           description  term_size  query_size  \\\n",
       "0    \"A vesicle that is released into the extracell...       2109         230   \n",
       "1    \"Any vesicle that is part of the extracellular...       2133         230   \n",
       "2    \"Organized structure of distinctive morphology...       2134         230   \n",
       "3    \"Organized structure of distinctive morphology...       2134         230   \n",
       "4    \"That part of a multicellular organism outside...       3303         230   \n",
       "..                                                 ...        ...         ...   \n",
       "184              skin 2; cells in granular layer[High]        248         176   \n",
       "185               skin 2; cells in spinous layer[High]        248         176   \n",
       "186  \"Any process of the immune system that execute...        698         227   \n",
       "187  \"Catalysis of the reaction: 4-aminobutanal + N...          2         223   \n",
       "188  \"Catalysis of the phosphorylation of tyrosine ...          2         223   \n",
       "\n",
       "     intersection_size  effective_domain_size  precision    recall    query  \\\n",
       "0                  120                  22097   0.521739  0.056899  query_1   \n",
       "1                  120                  22097   0.521739  0.056259  query_1   \n",
       "2                  120                  22097   0.521739  0.056232  query_1   \n",
       "3                  120                  22097   0.521739  0.056232  query_1   \n",
       "4                  134                  22097   0.582609  0.040569  query_1   \n",
       "..                 ...                    ...        ...       ...      ...   \n",
       "184                 13                  11019   0.073864  0.052419  query_1   \n",
       "185                 13                  11019   0.073864  0.052419  query_1   \n",
       "186                 21                  21031   0.092511  0.030086  query_1   \n",
       "187                  2                  20212   0.008969  1.000000  query_1   \n",
       "188                  2                  20212   0.008969  1.000000  query_1   \n",
       "\n",
       "                      parents  \n",
       "0    [GO:0005615, GO:1903561]  \n",
       "1    [GO:0031982, GO:0065010]  \n",
       "2    [GO:0005576, GO:0043226]  \n",
       "3    [GO:0043227, GO:0043230]  \n",
       "4    [GO:0005576, GO:0110165]  \n",
       "..                        ...  \n",
       "184             [HPA:0470182]  \n",
       "185             [HPA:0471412]  \n",
       "186              [GO:0002376]  \n",
       "187              [GO:0004029]  \n",
       "188              [GO:0004708]  \n",
       "\n",
       "[189 rows x 14 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from gprofiler import GProfiler\n",
    "\n",
    "# Initialize g:Profiler\n",
    "gp = GProfiler(return_dataframe=True)\n",
    "\n",
    "# Dictionary to store GO terms and p-values for each tissue\n",
    "go_dict = {}\n",
    "\n",
    "\n",
    "results = gp.profile(organism='hsapiens', query=list(subset_proteins.columns), sources=['GO:BP', 'GO:MF', 'GO:CC', 'HPA'])\n",
    "results = results[results['p_value']< 0.05]\n",
    "results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Establish whole blood reference proteome to use as a baseline; data from: https://onlinelibrary.wiley.com/doi/10.1111/trf.17254\n",
    "\n",
    "\n",
    "==> probably plasma, wait for Laura's whole blood samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UID</th>\n",
       "      <th>DIS_III_100</th>\n",
       "      <th>DIS_III_101</th>\n",
       "      <th>DIS_III_103</th>\n",
       "      <th>DIS_III_104</th>\n",
       "      <th>DIS_III_105</th>\n",
       "      <th>DIS_III_106</th>\n",
       "      <th>DIS_III_107</th>\n",
       "      <th>DIS_III_108</th>\n",
       "      <th>DIS_III_111</th>\n",
       "      <th>...</th>\n",
       "      <th>DIS_III_88</th>\n",
       "      <th>DIS_III_89</th>\n",
       "      <th>DIS_III_9</th>\n",
       "      <th>DIS_III_90</th>\n",
       "      <th>DIS_III_91</th>\n",
       "      <th>DIS_III_92</th>\n",
       "      <th>DIS_III_93</th>\n",
       "      <th>DIS_III_97</th>\n",
       "      <th>DIS_III_98</th>\n",
       "      <th>Gene_names</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UID1</td>\n",
       "      <td>32.161499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.333781</td>\n",
       "      <td>31.162190</td>\n",
       "      <td>31.897641</td>\n",
       "      <td>31.004103</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.454101</td>\n",
       "      <td>31.386228</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.781990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.375859</td>\n",
       "      <td>31.0944</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SERPINA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UID3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.230135</td>\n",
       "      <td>28.921956</td>\n",
       "      <td>27.817387</td>\n",
       "      <td>27.709186</td>\n",
       "      <td>28.044581</td>\n",
       "      <td>29.368676</td>\n",
       "      <td>28.907388</td>\n",
       "      <td>29.011693</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.720189</td>\n",
       "      <td>29.067124</td>\n",
       "      <td>27.971201</td>\n",
       "      <td>28.703685</td>\n",
       "      <td>27.364194</td>\n",
       "      <td>29.462716</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.858635</td>\n",
       "      <td>IGLV4-69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UID4</td>\n",
       "      <td>29.046976</td>\n",
       "      <td>29.094186</td>\n",
       "      <td>29.389483</td>\n",
       "      <td>27.779879</td>\n",
       "      <td>28.842571</td>\n",
       "      <td>28.408396</td>\n",
       "      <td>29.666126</td>\n",
       "      <td>30.562837</td>\n",
       "      <td>28.634676</td>\n",
       "      <td>...</td>\n",
       "      <td>28.754936</td>\n",
       "      <td>29.356659</td>\n",
       "      <td>28.402593</td>\n",
       "      <td>27.184329</td>\n",
       "      <td>28.805812</td>\n",
       "      <td>27.689725</td>\n",
       "      <td>30.632962</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.238838</td>\n",
       "      <td>IGLV8-61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UID5</td>\n",
       "      <td>24.427136</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.584629</td>\n",
       "      <td>25.357900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.305008</td>\n",
       "      <td>25.523338</td>\n",
       "      <td>...</td>\n",
       "      <td>25.314512</td>\n",
       "      <td>25.159926</td>\n",
       "      <td>25.241945</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>IGLV4-60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>UID6</td>\n",
       "      <td>24.461764</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.120681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.993885</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.371923</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.742937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.550306</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.066098</td>\n",
       "      <td>IGLV10-54</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 78 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    UID  DIS_III_100  DIS_III_101  DIS_III_103  DIS_III_104  DIS_III_105  \\\n",
       "0  UID1    32.161499          NaN          NaN    31.333781    31.162190   \n",
       "1  UID3          NaN    27.230135    28.921956    27.817387    27.709186   \n",
       "2  UID4    29.046976    29.094186    29.389483    27.779879    28.842571   \n",
       "3  UID5    24.427136          NaN    26.584629    25.357900          NaN   \n",
       "4  UID6    24.461764          NaN    26.120681          NaN    26.993885   \n",
       "\n",
       "   DIS_III_106  DIS_III_107  DIS_III_108  DIS_III_111  ...  DIS_III_88  \\\n",
       "0    31.897641    31.004103          NaN          NaN  ...         NaN   \n",
       "1    28.044581    29.368676    28.907388    29.011693  ...         NaN   \n",
       "2    28.408396    29.666126    30.562837    28.634676  ...   28.754936   \n",
       "3          NaN          NaN    25.305008    25.523338  ...   25.314512   \n",
       "4          NaN          NaN    25.371923          NaN  ...         NaN   \n",
       "\n",
       "   DIS_III_89  DIS_III_9  DIS_III_90  DIS_III_91  DIS_III_92  DIS_III_93  \\\n",
       "0   31.454101  31.386228         NaN   30.781990         NaN   30.375859   \n",
       "1   28.720189  29.067124   27.971201   28.703685   27.364194   29.462716   \n",
       "2   29.356659  28.402593   27.184329   28.805812   27.689725   30.632962   \n",
       "3   25.159926  25.241945         NaN         NaN         NaN         NaN   \n",
       "4   25.742937        NaN         NaN   25.550306         NaN         NaN   \n",
       "\n",
       "   DIS_III_97  DIS_III_98  Gene_names  \n",
       "0     31.0944         NaN    SERPINA1  \n",
       "1         NaN   27.858635    IGLV4-69  \n",
       "2         NaN   29.238838    IGLV8-61  \n",
       "3         NaN         NaN    IGLV4-60  \n",
       "4         NaN   27.066098   IGLV10-54  \n",
       "\n",
       "[5 rows x 78 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel('trf17254-sup-0003-tables1.xlsx', sheet_name='Quantified_DIS_III')\n",
    "gene_to_protein_map = pd.read_csv('idmapping_reviewed_true_2024_12_06.tsv', sep='\\t')   \n",
    "gene_to_protein_dict = dict(zip(gene_to_protein_map['From'], gene_to_protein_map['Entry']))\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DIS_III_100</th>\n",
       "      <th>DIS_III_101</th>\n",
       "      <th>DIS_III_103</th>\n",
       "      <th>DIS_III_104</th>\n",
       "      <th>DIS_III_105</th>\n",
       "      <th>DIS_III_106</th>\n",
       "      <th>DIS_III_107</th>\n",
       "      <th>DIS_III_108</th>\n",
       "      <th>DIS_III_111</th>\n",
       "      <th>DIS_III_112</th>\n",
       "      <th>...</th>\n",
       "      <th>DIS_III_87</th>\n",
       "      <th>DIS_III_88</th>\n",
       "      <th>DIS_III_89</th>\n",
       "      <th>DIS_III_9</th>\n",
       "      <th>DIS_III_90</th>\n",
       "      <th>DIS_III_91</th>\n",
       "      <th>DIS_III_92</th>\n",
       "      <th>DIS_III_93</th>\n",
       "      <th>DIS_III_97</th>\n",
       "      <th>DIS_III_98</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Entry</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P01009</th>\n",
       "      <td>32.161499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.333781</td>\n",
       "      <td>31.162190</td>\n",
       "      <td>31.897641</td>\n",
       "      <td>31.004103</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.528737</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.454101</td>\n",
       "      <td>31.386228</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.781990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30.375859</td>\n",
       "      <td>31.0944</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A0A075B6H9</th>\n",
       "      <td>NaN</td>\n",
       "      <td>27.230135</td>\n",
       "      <td>28.921956</td>\n",
       "      <td>27.817387</td>\n",
       "      <td>27.709186</td>\n",
       "      <td>28.044581</td>\n",
       "      <td>29.368676</td>\n",
       "      <td>28.907388</td>\n",
       "      <td>29.011693</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.720189</td>\n",
       "      <td>29.067124</td>\n",
       "      <td>27.971201</td>\n",
       "      <td>28.703685</td>\n",
       "      <td>27.364194</td>\n",
       "      <td>29.462716</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.858635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A0A075B6I0</th>\n",
       "      <td>29.046976</td>\n",
       "      <td>29.094186</td>\n",
       "      <td>29.389483</td>\n",
       "      <td>27.779879</td>\n",
       "      <td>28.842571</td>\n",
       "      <td>28.408396</td>\n",
       "      <td>29.666126</td>\n",
       "      <td>30.562837</td>\n",
       "      <td>28.634676</td>\n",
       "      <td>27.194227</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.754936</td>\n",
       "      <td>29.356659</td>\n",
       "      <td>28.402593</td>\n",
       "      <td>27.184329</td>\n",
       "      <td>28.805812</td>\n",
       "      <td>27.689725</td>\n",
       "      <td>30.632962</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.238838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A0A075B6I1</th>\n",
       "      <td>24.427136</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.584629</td>\n",
       "      <td>25.357900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.305008</td>\n",
       "      <td>25.523338</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>25.351377</td>\n",
       "      <td>25.314512</td>\n",
       "      <td>25.159926</td>\n",
       "      <td>25.241945</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A0A075B6I4</th>\n",
       "      <td>24.461764</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.120681</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.993885</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.371923</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.184877</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.742937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.550306</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27.066098</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 76 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            DIS_III_100  DIS_III_101  DIS_III_103  DIS_III_104  DIS_III_105  \\\n",
       "Entry                                                                         \n",
       "P01009        32.161499          NaN          NaN    31.333781    31.162190   \n",
       "A0A075B6H9          NaN    27.230135    28.921956    27.817387    27.709186   \n",
       "A0A075B6I0    29.046976    29.094186    29.389483    27.779879    28.842571   \n",
       "A0A075B6I1    24.427136          NaN    26.584629    25.357900          NaN   \n",
       "A0A075B6I4    24.461764          NaN    26.120681          NaN    26.993885   \n",
       "\n",
       "            DIS_III_106  DIS_III_107  DIS_III_108  DIS_III_111  DIS_III_112  \\\n",
       "Entry                                                                         \n",
       "P01009        31.897641    31.004103          NaN          NaN    31.528737   \n",
       "A0A075B6H9    28.044581    29.368676    28.907388    29.011693          NaN   \n",
       "A0A075B6I0    28.408396    29.666126    30.562837    28.634676    27.194227   \n",
       "A0A075B6I1          NaN          NaN    25.305008    25.523338          NaN   \n",
       "A0A075B6I4          NaN          NaN    25.371923          NaN    25.184877   \n",
       "\n",
       "            ...  DIS_III_87  DIS_III_88  DIS_III_89  DIS_III_9  DIS_III_90  \\\n",
       "Entry       ...                                                              \n",
       "P01009      ...         NaN         NaN   31.454101  31.386228         NaN   \n",
       "A0A075B6H9  ...         NaN         NaN   28.720189  29.067124   27.971201   \n",
       "A0A075B6I0  ...         NaN   28.754936   29.356659  28.402593   27.184329   \n",
       "A0A075B6I1  ...   25.351377   25.314512   25.159926  25.241945         NaN   \n",
       "A0A075B6I4  ...         NaN         NaN   25.742937        NaN         NaN   \n",
       "\n",
       "            DIS_III_91  DIS_III_92  DIS_III_93  DIS_III_97  DIS_III_98  \n",
       "Entry                                                                   \n",
       "P01009       30.781990         NaN   30.375859     31.0944         NaN  \n",
       "A0A075B6H9   28.703685   27.364194   29.462716         NaN   27.858635  \n",
       "A0A075B6I0   28.805812   27.689725   30.632962         NaN   29.238838  \n",
       "A0A075B6I1         NaN         NaN         NaN         NaN         NaN  \n",
       "A0A075B6I4   25.550306         NaN         NaN         NaN   27.066098  \n",
       "\n",
       "[5 rows x 76 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Entry'] = df['Gene_names'].map(gene_to_protein_dict)\n",
    "df.drop(columns=['UID', 'Gene_names'], inplace=True)\n",
    "df.set_index('Entry', inplace=True)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4271 model features are not in the sample and were added as zero values\n",
      "Warning: 100 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4272 model features are not in the sample and were added as zero values\n",
      "Warning: 103 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4269 model features are not in the sample and were added as zero values\n",
      "Warning: 108 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4269 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4268 model features are not in the sample and were added as zero values\n",
      "Warning: 104 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4267 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 99 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4279 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4271 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4267 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4278 model features are not in the sample and were added as zero values\n",
      "Warning: 105 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4278 model features are not in the sample and were added as zero values\n",
      "Warning: 103 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4271 model features are not in the sample and were added as zero values\n",
      "Warning: 93 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4269 model features are not in the sample and were added as zero values\n",
      "Warning: 99 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4270 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4277 model features are not in the sample and were added as zero values\n",
      "Warning: 96 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4271 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4271 model features are not in the sample and were added as zero values\n",
      "Warning: 99 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4269 model features are not in the sample and were added as zero values\n",
      "Warning: 100 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4271 model features are not in the sample and were added as zero values\n",
      "Warning: 104 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4284 model features are not in the sample and were added as zero values\n",
      "Warning: 99 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4267 model features are not in the sample and were added as zero values\n",
      "Warning: 105 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4270 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4272 model features are not in the sample and were added as zero values\n",
      "Warning: 105 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4276 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 94 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 96 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4385 and should be 4384\n",
      "4385 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4269 model features are not in the sample and were added as zero values\n",
      "Warning: 106 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4268 model features are not in the sample and were added as zero values\n",
      "Warning: 100 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4281 model features are not in the sample and were added as zero values\n",
      "Warning: 94 sample features are not in the model and were removed\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4268 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4271 model features are not in the sample and were added as zero values\n",
      "Warning: 97 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4267 model features are not in the sample and were added as zero values\n",
      "Warning: 106 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4270 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4272 model features are not in the sample and were added as zero values\n",
      "Warning: 97 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4272 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 100 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4278 model features are not in the sample and were added as zero values\n",
      "Warning: 99 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4274 model features are not in the sample and were added as zero values\n",
      "Warning: 97 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4273 model features are not in the sample and were added as zero values\n",
      "Warning: 104 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4271 model features are not in the sample and were added as zero values\n",
      "Warning: 100 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4276 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 104 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4273 model features are not in the sample and were added as zero values\n",
      "Warning: 99 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4271 model features are not in the sample and were added as zero values\n",
      "Warning: 102 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4279 model features are not in the sample and were added as zero values\n",
      "Warning: 94 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4274 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4273 model features are not in the sample and were added as zero values\n",
      "Warning: 103 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4279 model features are not in the sample and were added as zero values\n",
      "Warning: 97 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4276 model features are not in the sample and were added as zero values\n",
      "Warning: 102 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4272 model features are not in the sample and were added as zero values\n",
      "Warning: 99 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4269 model features are not in the sample and were added as zero values\n",
      "Warning: 102 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4277 model features are not in the sample and were added as zero values\n",
      "Warning: 99 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4276 model features are not in the sample and were added as zero values\n",
      "Warning: 93 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4267 model features are not in the sample and were added as zero values\n",
      "Warning: 100 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4283 model features are not in the sample and were added as zero values\n",
      "Warning: 92 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4385 and should be 4384\n",
      "4385 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 102 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4386 and should be 4384\n",
      "4386 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4276 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4385 and should be 4384\n",
      "4385 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4273 model features are not in the sample and were added as zero values\n",
      "Warning: 106 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4274 model features are not in the sample and were added as zero values\n",
      "Warning: 103 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4279 model features are not in the sample and were added as zero values\n",
      "Warning: 95 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4385 and should be 4384\n",
      "4385 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4272 model features are not in the sample and were added as zero values\n",
      "Warning: 102 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4278 model features are not in the sample and were added as zero values\n",
      "Warning: 105 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 101 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4281 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4276 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4273 model features are not in the sample and were added as zero values\n",
      "Warning: 102 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4281 model features are not in the sample and were added as zero values\n",
      "Warning: 96 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 97 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4273 model features are not in the sample and were added as zero values\n",
      "Warning: 103 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4387 and should be 4384\n",
      "4387 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4273 model features are not in the sample and were added as zero values\n",
      "Warning: 98 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4285 model features are not in the sample and were added as zero values\n",
      "Warning: 94 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4389 and should be 4384\n",
      "4389 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4275 model features are not in the sample and were added as zero values\n",
      "Warning: 96 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n",
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 4270 model features are not in the sample and were added as zero values\n",
      "Warning: 99 sample features are not in the model and were removed\n",
      "Error: Sample columns do not match model features\n",
      "Error: Features are 4388 and should be 4384\n",
      "4388 4384\n"
     ]
    }
   ],
   "source": [
    "#one by one, column by column\n",
    "prediction_dict={}\n",
    "for i in range(len(df.columns)):\n",
    "    subset = df.iloc[:, i].dropna()\n",
    "    subset_minmax = pd.DataFrame((subset - subset.min()) / (subset.max() - subset.min())).T\n",
    "    test = MLMarker(subset_minmax.iloc[0:1,:], binary = False, dev=False)\n",
    "    prediction = MLMarker.adjusted_absent_shap_values_df(test, penalty_factor=1)\n",
    "    prediction_dict[i] = prediction.sum(axis=1).head(3)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{0: tissue\n",
       " Monocytes         0.066564\n",
       " Adipose tissue    0.111495\n",
       " B-cells          -0.010195\n",
       " dtype: float64,\n",
       " 1: tissue\n",
       " Adipose tissue    0.124522\n",
       " Monocytes         0.072472\n",
       " B-cells          -0.010516\n",
       " dtype: float64,\n",
       " 2: tissue\n",
       " Adipose tissue    0.123428\n",
       " Monocytes         0.080162\n",
       " B-cells          -0.006979\n",
       " dtype: float64,\n",
       " 3: tissue\n",
       " Adipose tissue    0.120756\n",
       " Monocytes         0.072024\n",
       " B-cells          -0.010131\n",
       " dtype: float64,\n",
       " 4: tissue\n",
       " Monocytes         0.083499\n",
       " Adipose tissue    0.114986\n",
       " B-cells          -0.004891\n",
       " dtype: float64,\n",
       " 5: tissue\n",
       " Adipose tissue    0.127056\n",
       " Monocytes         0.072031\n",
       " B-cells          -0.010210\n",
       " dtype: float64,\n",
       " 6: tissue\n",
       " Monocytes         0.083597\n",
       " Adipose tissue    0.111864\n",
       " B-cells          -0.005878\n",
       " dtype: float64,\n",
       " 7: tissue\n",
       " Adipose tissue    0.133581\n",
       " Monocytes         0.073051\n",
       " B-cells          -0.007680\n",
       " dtype: float64,\n",
       " 8: tissue\n",
       " Adipose tissue    0.125362\n",
       " Monocytes         0.072375\n",
       " B-cells          -0.010263\n",
       " dtype: float64,\n",
       " 9: tissue\n",
       " Monocytes         0.075258\n",
       " Adipose tissue    0.110807\n",
       " B-cells          -0.008914\n",
       " dtype: float64,\n",
       " 10: tissue\n",
       " Adipose tissue    0.130431\n",
       " Monocytes         0.082233\n",
       " B-cells          -0.006179\n",
       " dtype: float64,\n",
       " 11: tissue\n",
       " Monocytes         0.084457\n",
       " Adipose tissue    0.116659\n",
       " B-cells          -0.004626\n",
       " dtype: float64,\n",
       " 12: tissue\n",
       " Adipose tissue    0.123391\n",
       " Monocytes         0.080164\n",
       " B-cells          -0.006979\n",
       " dtype: float64,\n",
       " 13: tissue\n",
       " Monocytes         0.075303\n",
       " Adipose tissue    0.111957\n",
       " B-cells          -0.008809\n",
       " dtype: float64,\n",
       " 14: tissue\n",
       " Adipose tissue    0.133962\n",
       " Monocytes         0.082297\n",
       " B-cells          -0.005663\n",
       " dtype: float64,\n",
       " 15: tissue\n",
       " Adipose tissue    0.130376\n",
       " Monocytes         0.081212\n",
       " B-cells          -0.006659\n",
       " dtype: float64,\n",
       " 16: tissue\n",
       " Adipose tissue    0.120902\n",
       " Monocytes         0.072266\n",
       " B-cells          -0.010687\n",
       " dtype: float64,\n",
       " 17: tissue\n",
       " Monocytes         0.075441\n",
       " Adipose tissue    0.108799\n",
       " B-cells          -0.008442\n",
       " dtype: float64,\n",
       " 18: tissue\n",
       " Adipose tissue    0.122769\n",
       " Monocytes         0.073508\n",
       " B-cells          -0.008693\n",
       " dtype: float64,\n",
       " 19: tissue\n",
       " Adipose tissue    0.131215\n",
       " Monocytes         0.082282\n",
       " B-cells          -0.005634\n",
       " dtype: float64,\n",
       " 20: tissue\n",
       " Adipose tissue    0.122974\n",
       " Monocytes         0.003280\n",
       " B-cells           0.000005\n",
       " dtype: float64,\n",
       " 21: tissue\n",
       " Monocytes         0.075167\n",
       " Adipose tissue    0.108163\n",
       " B-cells          -0.008599\n",
       " dtype: float64,\n",
       " 22: tissue\n",
       " Monocytes         0.085731\n",
       " Adipose tissue    0.116020\n",
       " B-cells          -0.004138\n",
       " dtype: float64,\n",
       " 23: tissue\n",
       " Monocytes         0.085323\n",
       " Adipose tissue    0.118312\n",
       " B-cells          -0.003897\n",
       " dtype: float64,\n",
       " 24: tissue\n",
       " Monocytes         0.084468\n",
       " Adipose tissue    0.117711\n",
       " B-cells          -0.004693\n",
       " dtype: float64,\n",
       " 25: tissue\n",
       " Monocytes         0.076079\n",
       " Adipose tissue    0.110383\n",
       " B-cells          -0.008659\n",
       " dtype: float64,\n",
       " 26: tissue\n",
       " Adipose tissue    0.128729\n",
       " Monocytes         0.081704\n",
       " B-cells          -0.006039\n",
       " dtype: float64,\n",
       " 27: tissue\n",
       " Adipose tissue    0.123640\n",
       " Monocytes         0.071998\n",
       " B-cells          -0.010155\n",
       " dtype: float64,\n",
       " 28: tissue\n",
       " Adipose tissue    0.131122\n",
       " Monocytes         0.081596\n",
       " B-cells          -0.007018\n",
       " dtype: float64,\n",
       " 29: tissue\n",
       " Adipose tissue    0.116502\n",
       " Monocytes         0.055748\n",
       " B-cells          -0.008367\n",
       " dtype: float64,\n",
       " 30: tissue\n",
       " Monocytes         0.083410\n",
       " Adipose tissue    0.113427\n",
       " B-cells          -0.005859\n",
       " dtype: float64,\n",
       " 31: tissue\n",
       " Adipose tissue    0.120743\n",
       " Monocytes         0.072024\n",
       " B-cells          -0.010131\n",
       " dtype: float64,\n",
       " 32: tissue\n",
       " Adipose tissue    0.122843\n",
       " Monocytes         0.071952\n",
       " B-cells          -0.010248\n",
       " dtype: float64,\n",
       " 33: tissue\n",
       " Adipose tissue    0.122662\n",
       " Monocytes         0.072948\n",
       " B-cells          -0.009306\n",
       " dtype: float64,\n",
       " 34: tissue\n",
       " Monocytes         0.083448\n",
       " Adipose tissue    0.108658\n",
       " B-cells          -0.005717\n",
       " dtype: float64,\n",
       " 35: tissue\n",
       " Adipose tissue    0.134850\n",
       " Monocytes        -0.001642\n",
       " B-cells          -0.007358\n",
       " dtype: float64,\n",
       " 36: tissue\n",
       " Adipose tissue    0.126818\n",
       " Monocytes         0.060724\n",
       " B-cells          -0.008132\n",
       " dtype: float64,\n",
       " 37: tissue\n",
       " Adipose tissue    0.131700\n",
       " Monocytes         0.062218\n",
       " B-cells          -0.007708\n",
       " dtype: float64,\n",
       " 38: tissue\n",
       " Adipose tissue    0.126685\n",
       " Monocytes         0.056827\n",
       " B-cells          -0.012128\n",
       " dtype: float64,\n",
       " 39: tissue\n",
       " Monocytes         0.075143\n",
       " Adipose tissue    0.105427\n",
       " B-cells          -0.009270\n",
       " dtype: float64,\n",
       " 40: tissue\n",
       " Monocytes         0.083133\n",
       " Adipose tissue    0.108405\n",
       " B-cells          -0.005488\n",
       " dtype: float64,\n",
       " 41: tissue\n",
       " Adipose tissue    0.123994\n",
       " Monocytes         0.056170\n",
       " B-cells          -0.012333\n",
       " dtype: float64,\n",
       " 42: tissue\n",
       " Adipose tissue    0.121297\n",
       " Monocytes         0.072565\n",
       " B-cells          -0.010239\n",
       " dtype: float64,\n",
       " 43: tissue\n",
       " Monocytes         0.075922\n",
       " Adipose tissue    0.109846\n",
       " B-cells          -0.008448\n",
       " dtype: float64,\n",
       " 44: tissue\n",
       " Adipose tissue    0.122171\n",
       " Monocytes         0.072632\n",
       " B-cells          -0.009592\n",
       " dtype: float64,\n",
       " 45: tissue\n",
       " Adipose tissue    0.131938\n",
       " Monocytes         0.081753\n",
       " B-cells          -0.005804\n",
       " dtype: float64,\n",
       " 46: tissue\n",
       " Monocytes         0.075392\n",
       " Adipose tissue    0.109532\n",
       " B-cells          -0.009169\n",
       " dtype: float64,\n",
       " 47: tissue\n",
       " Adipose tissue    0.130157\n",
       " Monocytes         0.060696\n",
       " B-cells          -0.008011\n",
       " dtype: float64,\n",
       " 48: tissue\n",
       " Adipose tissue    0.123926\n",
       " Monocytes         0.080644\n",
       " B-cells          -0.007407\n",
       " dtype: float64,\n",
       " 49: tissue\n",
       " Adipose tissue    0.131954\n",
       " Monocytes         0.080794\n",
       " B-cells          -0.007038\n",
       " dtype: float64,\n",
       " 50: tissue\n",
       " Adipose tissue    0.123988\n",
       " Monocytes         0.072966\n",
       " B-cells          -0.009321\n",
       " dtype: float64,\n",
       " 51: tissue\n",
       " Adipose tissue    0.124347\n",
       " Monocytes         0.021546\n",
       " B-cells           0.000442\n",
       " dtype: float64,\n",
       " 52: tissue\n",
       " Adipose tissue    0.124816\n",
       " Monocytes         0.073135\n",
       " B-cells          -0.009581\n",
       " dtype: float64,\n",
       " 53: tissue\n",
       " Adipose tissue    0.125986\n",
       " Monocytes         0.072092\n",
       " B-cells          -0.008753\n",
       " dtype: float64,\n",
       " 54: tissue\n",
       " Adipose tissue    0.126860\n",
       " Monocytes         0.081776\n",
       " B-cells          -0.006784\n",
       " dtype: float64,\n",
       " 55: tissue\n",
       " Adipose tissue    0.125353\n",
       " Monocytes         0.071994\n",
       " B-cells          -0.010194\n",
       " dtype: float64,\n",
       " 56: tissue\n",
       " Adipose tissue    0.127995\n",
       " Monocytes         0.055757\n",
       " B-cells          -0.012212\n",
       " dtype: float64,\n",
       " 57: tissue\n",
       " Monocytes         0.065712\n",
       " Adipose tissue    0.116166\n",
       " B-cells          -0.005839\n",
       " dtype: float64,\n",
       " 58: tissue\n",
       " Adipose tissue    0.127298\n",
       " Monocytes         0.062168\n",
       " B-cells          -0.007855\n",
       " dtype: float64,\n",
       " 59: tissue\n",
       " Adipose tissue    0.130175\n",
       " Monocytes         0.060993\n",
       " B-cells          -0.008367\n",
       " dtype: float64,\n",
       " 60: tissue\n",
       " Adipose tissue    0.122222\n",
       " Monocytes         0.072900\n",
       " B-cells          -0.009387\n",
       " dtype: float64,\n",
       " 61: tissue\n",
       " Adipose tissue    0.125010\n",
       " Monocytes         0.073150\n",
       " B-cells          -0.009737\n",
       " dtype: float64,\n",
       " 62: tissue\n",
       " Adipose tissue    0.120867\n",
       " Monocytes         0.073389\n",
       " B-cells          -0.009440\n",
       " dtype: float64,\n",
       " 63: tissue\n",
       " Adipose tissue    0.130537\n",
       " Monocytes         0.082616\n",
       " B-cells          -0.005797\n",
       " dtype: float64,\n",
       " 64: tissue\n",
       " Adipose tissue    0.129036\n",
       " Monocytes         0.081652\n",
       " B-cells          -0.006186\n",
       " dtype: float64,\n",
       " 65: tissue\n",
       " Adipose tissue    0.128795\n",
       " Monocytes         0.081057\n",
       " B-cells          -0.006920\n",
       " dtype: float64,\n",
       " 66: tissue\n",
       " Adipose tissue    0.128681\n",
       " Monocytes         0.004219\n",
       " B-cells           0.001212\n",
       " dtype: float64,\n",
       " 67: tissue\n",
       " Adipose tissue    0.127087\n",
       " Monocytes         0.074910\n",
       " B-cells          -0.009276\n",
       " dtype: float64,\n",
       " 68: tissue\n",
       " Adipose tissue    0.124154\n",
       " Monocytes         0.080210\n",
       " B-cells          -0.006886\n",
       " dtype: float64,\n",
       " 69: tissue\n",
       " Monocytes         0.077238\n",
       " Adipose tissue    0.106019\n",
       " B-cells          -0.006936\n",
       " dtype: float64,\n",
       " 70: tissue\n",
       " Adipose tissue    0.127456\n",
       " Monocytes         0.081758\n",
       " B-cells          -0.006830\n",
       " dtype: float64,\n",
       " 71: tissue\n",
       " Adipose tissue    0.131908\n",
       " Monocytes         0.081655\n",
       " B-cells          -0.006885\n",
       " dtype: float64,\n",
       " 72: tissue\n",
       " Monocytes         0.084286\n",
       " Adipose tissue    0.117272\n",
       " B-cells          -0.005197\n",
       " dtype: float64,\n",
       " 73: tissue\n",
       " Adipose tissue    0.132997\n",
       " Monocytes         0.003368\n",
       " B-cells           0.000332\n",
       " dtype: float64,\n",
       " 74: tissue\n",
       " Monocytes         0.075876\n",
       " Adipose tissue    0.106128\n",
       " B-cells          -0.008619\n",
       " dtype: float64,\n",
       " 75: tissue\n",
       " Adipose tissue    0.127550\n",
       " Monocytes         0.073378\n",
       " B-cells          -0.009587\n",
       " dtype: float64}"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prediction_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#row wise average to get average protein expression\n",
    "df['average'] = df.iloc[:,1:-2].mean(axis=1)\n",
    "baseline = df[['Entry', 'average']].pivot_table(index='Entry', values='average', aggfunc='mean')\n",
    "baseline['minmax_average'] = (baseline['average'] - baseline['average'].min()) / (baseline['average'].max() - baseline['average'].min())\n",
    "baseline_test = baseline[['minmax_average']].T\n",
    "\n",
    "test = MLMarker(baseline_test.iloc[0:1,:], binary = False, dev=False)\n",
    "prediction = MLMarker.adjusted_absent_shap_values_df(test, penalty_factor=1)\n",
    "prediction.sum(axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein</th>\n",
       "      <th>Protein ID</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Length</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Protein Description</th>\n",
       "      <th>Protein Existence</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Protein Probability</th>\n",
       "      <th>...</th>\n",
       "      <th>Razor Peptides</th>\n",
       "      <th>Total Spectral Count</th>\n",
       "      <th>Unique Spectral Count</th>\n",
       "      <th>Razor Spectral Count</th>\n",
       "      <th>Total Intensity</th>\n",
       "      <th>Unique Intensity</th>\n",
       "      <th>Razor Intensity</th>\n",
       "      <th>Razor Assigned Modifications</th>\n",
       "      <th>Razor Observed Modifications</th>\n",
       "      <th>Indistinguishable Proteins</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>QC|RT_Peptide|JPT_Retention_Time_Peptide</td>\n",
       "      <td>QC|RT_Peptide|JPT_Retention_Time_Peptide</td>\n",
       "      <td>QC|RT_Peptide|JPT_Retention_Time_Peptide</td>\n",
       "      <td>NaN</td>\n",
       "      <td>456</td>\n",
       "      <td>NaN</td>\n",
       "      <td>QC|RT_Peptide|JPT_Retention_Time_Peptide</td>\n",
       "      <td>NaN</td>\n",
       "      <td>72.37</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>31</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>65</td>\n",
       "      <td>5.655622e+09</td>\n",
       "      <td>5.655622e+09</td>\n",
       "      <td>5.655622e+09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rev_sp|A0JNW5|BLT3B_HUMAN</td>\n",
       "      <td>rev_A0JNW5</td>\n",
       "      <td>rev_BLT3B_HUMAN</td>\n",
       "      <td>rev_BLTP3B</td>\n",
       "      <td>1464</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Bridge-like lipid transfer protein family memb...</td>\n",
       "      <td>1:Experimental evidence at protein level</td>\n",
       "      <td>1.84</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>21M(15.9949), 4C(57.0214)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rev_sp|A4D0S4|LAMB4_HUMAN</td>\n",
       "      <td>rev_A4D0S4</td>\n",
       "      <td>rev_LAMB4_HUMAN</td>\n",
       "      <td>rev_LAMB4</td>\n",
       "      <td>1761</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Laminin subunit beta-4</td>\n",
       "      <td>2:Experimental evidence at transcript level</td>\n",
       "      <td>0.68</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>10C(57.0214)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rev_sp|A5YKK6|CNOT1_HUMAN</td>\n",
       "      <td>rev_A5YKK6</td>\n",
       "      <td>rev_CNOT1_HUMAN</td>\n",
       "      <td>rev_CNOT1</td>\n",
       "      <td>2376</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>CCR4-NOT transcription complex subunit 1</td>\n",
       "      <td>1:Experimental evidence at protein level</td>\n",
       "      <td>0.38</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rev_sp|A6H8Y1|BDP1_HUMAN</td>\n",
       "      <td>rev_A6H8Y1</td>\n",
       "      <td>rev_BDP1_HUMAN</td>\n",
       "      <td>rev_BDP1</td>\n",
       "      <td>2624</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Transcription factor TFIIIB component B'' homolog</td>\n",
       "      <td>1:Experimental evidence at protein level</td>\n",
       "      <td>0.46</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4.032473e+08</td>\n",
       "      <td>4.032473e+08</td>\n",
       "      <td>4.032473e+08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    Protein  \\\n",
       "0  QC|RT_Peptide|JPT_Retention_Time_Peptide   \n",
       "1                 rev_sp|A0JNW5|BLT3B_HUMAN   \n",
       "2                 rev_sp|A4D0S4|LAMB4_HUMAN   \n",
       "3                 rev_sp|A5YKK6|CNOT1_HUMAN   \n",
       "4                  rev_sp|A6H8Y1|BDP1_HUMAN   \n",
       "\n",
       "                                 Protein ID  \\\n",
       "0  QC|RT_Peptide|JPT_Retention_Time_Peptide   \n",
       "1                                rev_A0JNW5   \n",
       "2                                rev_A4D0S4   \n",
       "3                                rev_A5YKK6   \n",
       "4                                rev_A6H8Y1   \n",
       "\n",
       "                                 Entry Name        Gene  Length      Organism  \\\n",
       "0  QC|RT_Peptide|JPT_Retention_Time_Peptide         NaN     456           NaN   \n",
       "1                           rev_BLT3B_HUMAN  rev_BLTP3B    1464  Homo sapiens   \n",
       "2                           rev_LAMB4_HUMAN   rev_LAMB4    1761  Homo sapiens   \n",
       "3                           rev_CNOT1_HUMAN   rev_CNOT1    2376  Homo sapiens   \n",
       "4                            rev_BDP1_HUMAN    rev_BDP1    2624  Homo sapiens   \n",
       "\n",
       "                                 Protein Description  \\\n",
       "0           QC|RT_Peptide|JPT_Retention_Time_Peptide   \n",
       "1  Bridge-like lipid transfer protein family memb...   \n",
       "2                             Laminin subunit beta-4   \n",
       "3           CCR4-NOT transcription complex subunit 1   \n",
       "4  Transcription factor TFIIIB component B'' homolog   \n",
       "\n",
       "                             Protein Existence  Coverage  Protein Probability  \\\n",
       "0                                          NaN     72.37                  1.0   \n",
       "1     1:Experimental evidence at protein level      1.84                  1.0   \n",
       "2  2:Experimental evidence at transcript level      0.68                  1.0   \n",
       "3     1:Experimental evidence at protein level      0.38                  1.0   \n",
       "4     1:Experimental evidence at protein level      0.46                  1.0   \n",
       "\n",
       "   ...  Razor Peptides  Total Spectral Count  Unique Spectral Count  \\\n",
       "0  ...              31                    65                     65   \n",
       "1  ...               1                     1                      1   \n",
       "2  ...               1                     1                      1   \n",
       "3  ...               1                     1                      1   \n",
       "4  ...               1                     2                      2   \n",
       "\n",
       "   Razor Spectral Count  Total Intensity  Unique Intensity  Razor Intensity  \\\n",
       "0                    65     5.655622e+09      5.655622e+09     5.655622e+09   \n",
       "1                     1     0.000000e+00      0.000000e+00     0.000000e+00   \n",
       "2                     1     0.000000e+00      0.000000e+00     0.000000e+00   \n",
       "3                     1     0.000000e+00      0.000000e+00     0.000000e+00   \n",
       "4                     2     4.032473e+08      4.032473e+08     4.032473e+08   \n",
       "\n",
       "   Razor Assigned Modifications  Razor Observed Modifications  \\\n",
       "0                           NaN                           NaN   \n",
       "1     21M(15.9949), 4C(57.0214)                           NaN   \n",
       "2                  10C(57.0214)                           NaN   \n",
       "3                           NaN                           NaN   \n",
       "4                           NaN                           NaN   \n",
       "\n",
       "   Indistinguishable Proteins  \n",
       "0                         NaN  \n",
       "1                         NaN  \n",
       "2                         NaN  \n",
       "3                         NaN  \n",
       "4                         NaN  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('/home/compomics/git/MLMarker/Projects/protein(3).tsv', sep='\\t')  \n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Total Intensity</th>\n",
       "      <th>Protein ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.006154</td>\n",
       "      <td>QC|RT_Peptide|JPT_Retention_Time_Peptide</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>0.000640</td>\n",
       "      <td>A0A075B6H7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>0.000015</td>\n",
       "      <td>A0A087WSY6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>0.000011</td>\n",
       "      <td>A0A0A0MS15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>0.000184</td>\n",
       "      <td>A0A0B4J1V0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Total Intensity                                Protein ID\n",
       "0           0.006154  QC|RT_Peptide|JPT_Retention_Time_Peptide\n",
       "195         0.000640                                A0A075B6H7\n",
       "196         0.000015                                A0A087WSY6\n",
       "197         0.000011                                A0A0A0MS15\n",
       "198         0.000184                                A0A0B4J1V0"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df[~df['Protein ID'].str.contains('rev')]\n",
    "df = df[['Protein ID', 'Total Intensity']]\n",
    "dfminmax = pd.DataFrame((df['Total Intensity'] - df['Total Intensity'].min()) / (df['Total Intensity'].max() - df['Total Intensity'].min()))\n",
    "dfminmax['Protein ID'] = df['Protein ID']\n",
    "dfminmax.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Protein ID</th>\n",
       "      <th>A0A024RBG1</th>\n",
       "      <th>A0A075B6H7</th>\n",
       "      <th>A0A075B6K5</th>\n",
       "      <th>A0A075B6P5</th>\n",
       "      <th>A0A075B6R2</th>\n",
       "      <th>A0A075B6S2</th>\n",
       "      <th>A0A075B759</th>\n",
       "      <th>A0A075B767</th>\n",
       "      <th>A0A087WSX0</th>\n",
       "      <th>A0A087WSY4</th>\n",
       "      <th>...</th>\n",
       "      <th>Q9Y6U3</th>\n",
       "      <th>Q9Y6W3</th>\n",
       "      <th>Q9Y6W5</th>\n",
       "      <th>Q9Y6X0</th>\n",
       "      <th>Q9Y6X3</th>\n",
       "      <th>Q9Y6X4</th>\n",
       "      <th>Q9Y6X5</th>\n",
       "      <th>Q9Y6X8</th>\n",
       "      <th>Q9Y6Y8</th>\n",
       "      <th>QC|RT_Peptide|JPT_Retention_Time_Peptide</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Total Intensity</th>\n",
       "      <td>0.000309</td>\n",
       "      <td>0.00064</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>0.000409</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>0.000402</td>\n",
       "      <td>0.012899</td>\n",
       "      <td>0.012899</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000179</td>\n",
       "      <td>0.000087</td>\n",
       "      <td>0.000104</td>\n",
       "      <td>0.005356</td>\n",
       "      <td>0.000994</td>\n",
       "      <td>0.000372</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0.006154</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 5872 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Protein ID       A0A024RBG1  A0A075B6H7  A0A075B6K5  A0A075B6P5  A0A075B6R2  \\\n",
       "Total Intensity    0.000309     0.00064    0.000043    0.000409    0.000109   \n",
       "\n",
       "Protein ID       A0A075B6S2  A0A075B759  A0A075B767  A0A087WSX0  A0A087WSY4  \\\n",
       "Total Intensity    0.000402    0.012899    0.012899    0.000008    0.000109   \n",
       "\n",
       "Protein ID       ...    Q9Y6U3    Q9Y6W3    Q9Y6W5    Q9Y6X0    Q9Y6X3  \\\n",
       "Total Intensity  ...  0.000179  0.000087  0.000104  0.005356  0.000994   \n",
       "\n",
       "Protein ID         Q9Y6X4    Q9Y6X5    Q9Y6X8    Q9Y6Y8  \\\n",
       "Total Intensity  0.000372  0.000065  0.000064  0.000027   \n",
       "\n",
       "Protein ID       QC|RT_Peptide|JPT_Retention_Time_Peptide  \n",
       "Total Intensity                                  0.006154  \n",
       "\n",
       "[1 rows x 5872 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "baseline = dfminmax.pivot_table(index='Protein ID', values='Total Intensity', aggfunc='mean').T\n",
    "baseline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator DecisionTreeClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/compomics/miniconda3/envs/mlmarker/lib/python3.9/site-packages/sklearn/base.py:376: InconsistentVersionWarning: Trying to unpickle estimator RandomForestClassifier from version 1.5.1 when using version 1.4.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:\n",
      "https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: 1580 model features are not in the sample and were added as zero values\n",
      "Warning: 3068 sample features are not in the model and were removed\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A0A075B6H7</th>\n",
       "      <th>A0A075B6I0</th>\n",
       "      <th>A0A075B6Q5</th>\n",
       "      <th>A0A0A0MS14</th>\n",
       "      <th>A0A0A0MS15</th>\n",
       "      <th>A0A0B4J1U7</th>\n",
       "      <th>A0A0B4J1V0</th>\n",
       "      <th>A0A0B4J1V6</th>\n",
       "      <th>A0A0B4J1X8</th>\n",
       "      <th>A0A0B4J2F0</th>\n",
       "      <th>...</th>\n",
       "      <th>Q9Y6U3</th>\n",
       "      <th>Q9Y6V0</th>\n",
       "      <th>Q9Y6V7</th>\n",
       "      <th>Q9Y6W3</th>\n",
       "      <th>Q9Y6W5</th>\n",
       "      <th>Q9Y6X5</th>\n",
       "      <th>Q9Y6X8</th>\n",
       "      <th>Q9Y6X9</th>\n",
       "      <th>Q9Y6Y0</th>\n",
       "      <th>Q9Y6Y8</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tissue</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Brain</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.549587e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.993800e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.049445e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Skeletal muscle</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tonsil</th>\n",
       "      <td>5.702167e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-5.207505e-05</td>\n",
       "      <td>1.871994e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>9.869282e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.610759e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Liver</th>\n",
       "      <td>7.296142e-08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.241969e-07</td>\n",
       "      <td>1.525886e-06</td>\n",
       "      <td>2.262213e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.091532e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.668003e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.584245e-08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Parotid gland</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.130864e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.121666e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Heart</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.910446e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.501882e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.911859e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.866259e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Esophagus</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.794944e-06</td>\n",
       "      <td>1.253665e-06</td>\n",
       "      <td>-9.176483e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.362754e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-3.473670e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.898931e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Kidney</th>\n",
       "      <td>1.222897e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.819625e-06</td>\n",
       "      <td>1.568718e-05</td>\n",
       "      <td>1.327907e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.208483e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.323511e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>5.710612e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.354605e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Testis</th>\n",
       "      <td>4.906201e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.301708e-06</td>\n",
       "      <td>1.238322e-05</td>\n",
       "      <td>-6.455149e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-5.886075e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.491171e-07</td>\n",
       "      <td>1.443131e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.026059e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-2.924150e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ovary</th>\n",
       "      <td>2.301806e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.914878e-06</td>\n",
       "      <td>-9.405618e-07</td>\n",
       "      <td>3.158437e-07</td>\n",
       "      <td>5.949807e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.462327e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.282854e-06</td>\n",
       "      <td>4.629977e-08</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.174848e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.029453e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Placenta</th>\n",
       "      <td>8.030190e-08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.680409e-05</td>\n",
       "      <td>2.710627e-07</td>\n",
       "      <td>2.724743e-07</td>\n",
       "      <td>1.631753e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4.024489e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.977215e-07</td>\n",
       "      <td>2.934621e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.152396e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pituitary gland</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.072632e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.966329e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.052748e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.074350e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.666295e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Salivary gland</th>\n",
       "      <td>2.644418e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.522715e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.161036e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.436281e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-9.650798e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.431778e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lung</th>\n",
       "      <td>9.080319e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.472834e-06</td>\n",
       "      <td>-8.828466e-07</td>\n",
       "      <td>-8.950378e-07</td>\n",
       "      <td>8.511206e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.522107e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.162213e-07</td>\n",
       "      <td>1.446941e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.594664e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.471847e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Urinary bladder</th>\n",
       "      <td>3.693272e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.341737e-05</td>\n",
       "      <td>8.360682e-07</td>\n",
       "      <td>4.562187e-07</td>\n",
       "      <td>1.341663e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.964087e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.851462e-07</td>\n",
       "      <td>6.687744e-08</td>\n",
       "      <td>9.987759e-07</td>\n",
       "      <td>-2.250733e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.531908e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thyroid</th>\n",
       "      <td>3.047694e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.016400e-05</td>\n",
       "      <td>4.425253e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.151454e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.241559e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Adipose tissue</th>\n",
       "      <td>-2.239719e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.015252e-04</td>\n",
       "      <td>4.194013e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-7.467318e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.010680e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.841947e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.192275e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pancreas</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.002764e-06</td>\n",
       "      <td>3.114366e-06</td>\n",
       "      <td>2.415276e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.039069e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.029046e-07</td>\n",
       "      <td>1.089807e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-4.275914e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Appendix</th>\n",
       "      <td>-3.517630e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.074836e-05</td>\n",
       "      <td>4.170311e-06</td>\n",
       "      <td>8.902294e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.075687e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-6.155127e-06</td>\n",
       "      <td>2.820877e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.919145e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stomach</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.530616e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.579219e-07</td>\n",
       "      <td>1.625477e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.529882e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.914602e-07</td>\n",
       "      <td>-1.871132e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-3.418425e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Oviduct</th>\n",
       "      <td>8.253251e-08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.398631e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.980439e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.530465e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Spleen</th>\n",
       "      <td>3.584022e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.435086e-06</td>\n",
       "      <td>9.360698e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.024990e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.938531e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>8.025293e-08</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.162214e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.402550e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Duodenum</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.838953e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.352850e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.776144e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-3.672819e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-4.351394e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Endometrium</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.377271e-06</td>\n",
       "      <td>6.228731e-06</td>\n",
       "      <td>2.415276e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.903710e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.539071e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.961429e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-7.097229e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rectum</th>\n",
       "      <td>1.732834e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.831308e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.986040e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.621472e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.853947e-07</td>\n",
       "      <td>1.178844e-06</td>\n",
       "      <td>1.966340e-06</td>\n",
       "      <td>-3.739640e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.252390e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>B-cells</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.148445e-07</td>\n",
       "      <td>8.012065e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.238927e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-6.247138e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.603824e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nasal polyps</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-5.528119e-06</td>\n",
       "      <td>-1.892293e-05</td>\n",
       "      <td>-3.011632e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lymph node</th>\n",
       "      <td>1.556430e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.249638e-06</td>\n",
       "      <td>4.781415e-06</td>\n",
       "      <td>3.983722e-07</td>\n",
       "      <td>1.207497e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-3.492023e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.222434e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-2.092971e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.321402e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Monocytes</th>\n",
       "      <td>2.334645e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.085346e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.257634e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.918369e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Smooth muscle</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.699315e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.041559e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-5.070917e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>7.490819e-07</td>\n",
       "      <td>-1.284781e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.033501e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Prostate</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.002582e-06</td>\n",
       "      <td>5.474625e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5.911570e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>9.972075e-08</td>\n",
       "      <td>2.996328e-07</td>\n",
       "      <td>2.600362e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.045105e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Colon</th>\n",
       "      <td>1.341549e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.828555e-06</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.398086e-07</td>\n",
       "      <td>5.842727e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>9.193933e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.283558e-06</td>\n",
       "      <td>1.449836e-06</td>\n",
       "      <td>8.528214e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.162524e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Small intestine</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.072632e-06</td>\n",
       "      <td>6.316875e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.095658e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.639189e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.834132e-08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.930432e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Adrenal gland</th>\n",
       "      <td>1.061202e-07</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.642442e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.092818e-05</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.117195e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.190510e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NK-cells</th>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.342709e-07</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.329588e-06</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>-1.044446e-05</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35 rows × 4384 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   A0A075B6H7  A0A075B6I0  A0A075B6Q5  A0A0A0MS14  \\\n",
       "tissue                                                              \n",
       "Brain            0.000000e+00         0.0         0.0         0.0   \n",
       "Skeletal muscle  0.000000e+00         0.0         0.0         0.0   \n",
       "Tonsil           5.702167e-07         0.0         0.0         0.0   \n",
       "Liver            7.296142e-08         0.0         0.0         0.0   \n",
       "Parotid gland    0.000000e+00         0.0         0.0         0.0   \n",
       "Heart            0.000000e+00         0.0         0.0         0.0   \n",
       "Esophagus        0.000000e+00         0.0         0.0         0.0   \n",
       "Kidney           1.222897e-06         0.0         0.0         0.0   \n",
       "Testis           4.906201e-07         0.0         0.0         0.0   \n",
       "Ovary            2.301806e-07         0.0         0.0         0.0   \n",
       "Placenta         8.030190e-08         0.0         0.0         0.0   \n",
       "Pituitary gland  0.000000e+00         0.0         0.0         0.0   \n",
       "Salivary gland   2.644418e-07         0.0         0.0         0.0   \n",
       "Lung             9.080319e-07         0.0         0.0         0.0   \n",
       "Urinary bladder  3.693272e-07         0.0         0.0         0.0   \n",
       "Thyroid          3.047694e-07         0.0         0.0         0.0   \n",
       "Adipose tissue  -2.239719e-06         0.0         0.0         0.0   \n",
       "Pancreas         0.000000e+00         0.0         0.0         0.0   \n",
       "Appendix        -3.517630e-06         0.0         0.0         0.0   \n",
       "Stomach          0.000000e+00         0.0         0.0         0.0   \n",
       "Oviduct          8.253251e-08         0.0         0.0         0.0   \n",
       "Spleen           3.584022e-07         0.0         0.0         0.0   \n",
       "Duodenum         0.000000e+00         0.0         0.0         0.0   \n",
       "Endometrium      0.000000e+00         0.0         0.0         0.0   \n",
       "Rectum           1.732834e-07         0.0         0.0         0.0   \n",
       "B-cells          0.000000e+00         0.0         0.0         0.0   \n",
       "Nasal polyps     0.000000e+00         0.0         0.0         0.0   \n",
       "Lymph node       1.556430e-07         0.0         0.0         0.0   \n",
       "Monocytes        2.334645e-07         0.0         0.0         0.0   \n",
       "Smooth muscle    0.000000e+00         0.0         0.0         0.0   \n",
       "Prostate         0.000000e+00         0.0         0.0         0.0   \n",
       "Colon            1.341549e-07         0.0         0.0         0.0   \n",
       "Small intestine  0.000000e+00         0.0         0.0         0.0   \n",
       "Adrenal gland    1.061202e-07         0.0         0.0         0.0   \n",
       "NK-cells         0.000000e+00         0.0         0.0         0.0   \n",
       "\n",
       "                   A0A0A0MS15    A0A0B4J1U7    A0A0B4J1V0    A0A0B4J1V6  \\\n",
       "tissue                                                                    \n",
       "Brain            0.000000e+00  1.549587e-05  0.000000e+00  0.000000e+00   \n",
       "Skeletal muscle  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "Tonsil           0.000000e+00 -5.207505e-05  1.871994e-05  0.000000e+00   \n",
       "Liver            8.241969e-07  1.525886e-06  2.262213e-06  0.000000e+00   \n",
       "Parotid gland    7.130864e-06  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "Heart            0.000000e+00  0.000000e+00  3.910446e-07  0.000000e+00   \n",
       "Esophagus        4.794944e-06  1.253665e-06 -9.176483e-06  0.000000e+00   \n",
       "Kidney           1.819625e-06  1.568718e-05  1.327907e-06  0.000000e+00   \n",
       "Testis           1.301708e-06  1.238322e-05 -6.455149e-07  0.000000e+00   \n",
       "Ovary           -1.914878e-06 -9.405618e-07  3.158437e-07  5.949807e-07   \n",
       "Placenta         1.680409e-05  2.710627e-07  2.724743e-07  1.631753e-07   \n",
       "Pituitary gland  0.000000e+00  4.072632e-06  0.000000e+00  6.966329e-07   \n",
       "Salivary gland   3.522715e-06  0.000000e+00  2.161036e-07  0.000000e+00   \n",
       "Lung             1.472834e-06 -8.828466e-07 -8.950378e-07  8.511206e-07   \n",
       "Urinary bladder  1.341737e-05  8.360682e-07  4.562187e-07  1.341663e-06   \n",
       "Thyroid          0.000000e+00 -1.016400e-05  4.425253e-06  0.000000e+00   \n",
       "Adipose tissue  -1.015252e-04  4.194013e-07  0.000000e+00 -7.467318e-06   \n",
       "Pancreas         8.002764e-06  3.114366e-06  2.415276e-07  0.000000e+00   \n",
       "Appendix         4.074836e-05  4.170311e-06  8.902294e-07  0.000000e+00   \n",
       "Stomach         -1.530616e-05  0.000000e+00  1.579219e-07  1.625477e-06   \n",
       "Oviduct          9.398631e-07  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "Spleen           3.435086e-06  9.360698e-06  0.000000e+00  4.024990e-07   \n",
       "Duodenum         1.838953e-06  0.000000e+00 -1.352850e-06  0.000000e+00   \n",
       "Endometrium      2.377271e-06  6.228731e-06  2.415276e-07  0.000000e+00   \n",
       "Rectum           1.831308e-06  0.000000e+00  4.986040e-07  0.000000e+00   \n",
       "B-cells          0.000000e+00  3.148445e-07  8.012065e-06  0.000000e+00   \n",
       "Nasal polyps    -5.528119e-06 -1.892293e-05 -3.011632e-05  0.000000e+00   \n",
       "Lymph node       5.249638e-06  4.781415e-06  3.983722e-07  1.207497e-06   \n",
       "Monocytes        0.000000e+00  0.000000e+00  2.085346e-06  0.000000e+00   \n",
       "Smooth muscle    4.699315e-07  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "Prostate         0.000000e+00 -1.002582e-06  5.474625e-07  0.000000e+00   \n",
       "Colon            7.828555e-06  0.000000e+00 -1.398086e-07  5.842727e-07   \n",
       "Small intestine  0.000000e+00  4.072632e-06  6.316875e-07  0.000000e+00   \n",
       "Adrenal gland    4.642442e-07  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "NK-cells         0.000000e+00  0.000000e+00  2.342709e-07  0.000000e+00   \n",
       "\n",
       "                 A0A0B4J1X8  A0A0B4J2F0  ...        Q9Y6U3  Q9Y6V0  Q9Y6V7  \\\n",
       "tissue                                   ...                                 \n",
       "Brain                   0.0         0.0  ...  1.993800e-06     0.0     0.0   \n",
       "Skeletal muscle         0.0         0.0  ...  0.000000e+00     0.0     0.0   \n",
       "Tonsil                  0.0         0.0  ...  9.869282e-07     0.0     0.0   \n",
       "Liver                   0.0         0.0  ... -1.091532e-04     0.0     0.0   \n",
       "Parotid gland           0.0         0.0  ... -1.121666e-06     0.0     0.0   \n",
       "Heart                   0.0         0.0  ... -1.501882e-05     0.0     0.0   \n",
       "Esophagus               0.0         0.0  ...  1.362754e-05     0.0     0.0   \n",
       "Kidney                  0.0         0.0  ...  7.208483e-05     0.0     0.0   \n",
       "Testis                  0.0         0.0  ... -5.886075e-06     0.0     0.0   \n",
       "Ovary                   0.0         0.0  ... -1.462327e-05     0.0     0.0   \n",
       "Placenta                0.0         0.0  ...  4.024489e-05     0.0     0.0   \n",
       "Pituitary gland         0.0         0.0  ... -2.052748e-04     0.0     0.0   \n",
       "Salivary gland          0.0         0.0  ... -4.436281e-06     0.0     0.0   \n",
       "Lung                    0.0         0.0  ...  6.522107e-05     0.0     0.0   \n",
       "Urinary bladder         0.0         0.0  ... -1.964087e-05     0.0     0.0   \n",
       "Thyroid                 0.0         0.0  ... -3.151454e-06     0.0     0.0   \n",
       "Adipose tissue          0.0         0.0  ... -1.010680e-04     0.0     0.0   \n",
       "Pancreas                0.0         0.0  ... -1.039069e-05     0.0     0.0   \n",
       "Appendix                0.0         0.0  ... -1.075687e-04     0.0     0.0   \n",
       "Stomach                 0.0         0.0  ...  1.529882e-04     0.0     0.0   \n",
       "Oviduct                 0.0         0.0  ... -2.980439e-05     0.0     0.0   \n",
       "Spleen                  0.0         0.0  ... -2.938531e-05     0.0     0.0   \n",
       "Duodenum                0.0         0.0  ...  1.776144e-04     0.0     0.0   \n",
       "Endometrium             0.0         0.0  ... -3.903710e-05     0.0     0.0   \n",
       "Rectum                  0.0         0.0  ...  3.621472e-05     0.0     0.0   \n",
       "B-cells                 0.0         0.0  ... -4.238927e-06     0.0     0.0   \n",
       "Nasal polyps            0.0         0.0  ...  0.000000e+00     0.0     0.0   \n",
       "Lymph node              0.0         0.0  ... -3.492023e-05     0.0     0.0   \n",
       "Monocytes               0.0         0.0  ...  0.000000e+00     0.0     0.0   \n",
       "Smooth muscle           0.0         0.0  ... -2.041559e-05     0.0     0.0   \n",
       "Prostate                0.0         0.0  ...  5.911570e-06     0.0     0.0   \n",
       "Colon                   0.0         0.0  ...  9.193933e-05     0.0     0.0   \n",
       "Small intestine         0.0         0.0  ...  1.095658e-04     0.0     0.0   \n",
       "Adrenal gland           0.0         0.0  ... -1.092818e-05     0.0     0.0   \n",
       "NK-cells                0.0         0.0  ... -2.329588e-06     0.0     0.0   \n",
       "\n",
       "                       Q9Y6W3        Q9Y6W5        Q9Y6X5        Q9Y6X8  \\\n",
       "tissue                                                                    \n",
       "Brain            0.000000e+00  0.000000e+00  0.000000e+00  2.049445e-07   \n",
       "Skeletal muscle  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "Tonsil           0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "Liver            0.000000e+00  1.668003e-06  0.000000e+00  1.584245e-08   \n",
       "Parotid gland    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "Heart            0.000000e+00  4.911859e-06  0.000000e+00  0.000000e+00   \n",
       "Esophagus        0.000000e+00  0.000000e+00  0.000000e+00 -3.473670e-06   \n",
       "Kidney           0.000000e+00  1.323511e-05  0.000000e+00  5.710612e-07   \n",
       "Testis           1.491171e-07  1.443131e-06  0.000000e+00 -1.026059e-07   \n",
       "Ovary            2.282854e-06  4.629977e-08  0.000000e+00  1.174848e-06   \n",
       "Placenta         3.977215e-07  2.934621e-07  0.000000e+00  1.152396e-06   \n",
       "Pituitary gland  0.000000e+00  2.074350e-06  0.000000e+00  4.666295e-07   \n",
       "Salivary gland  -9.650798e-07  0.000000e+00  0.000000e+00  2.431778e-06   \n",
       "Lung             3.162213e-07  1.446941e-06  0.000000e+00  3.594664e-05   \n",
       "Urinary bladder  6.851462e-07  6.687744e-08  9.987759e-07 -2.250733e-06   \n",
       "Thyroid          0.000000e+00  2.241559e-06  0.000000e+00  0.000000e+00   \n",
       "Adipose tissue   0.000000e+00  1.841947e-06  0.000000e+00  1.192275e-07   \n",
       "Pancreas         5.029046e-07  1.089807e-05  0.000000e+00 -4.275914e-07   \n",
       "Appendix         0.000000e+00  0.000000e+00 -6.155127e-06  2.820877e-07   \n",
       "Stomach          0.000000e+00  0.000000e+00  6.914602e-07 -1.871132e-05   \n",
       "Oviduct          0.000000e+00  0.000000e+00  0.000000e+00  6.530465e-07   \n",
       "Spleen           0.000000e+00  8.025293e-08  0.000000e+00  4.162214e-06   \n",
       "Duodenum         0.000000e+00 -3.672819e-07  0.000000e+00 -4.351394e-07   \n",
       "Endometrium      2.539071e-06  0.000000e+00  0.000000e+00 -1.961429e-06   \n",
       "Rectum           3.853947e-07  1.178844e-06  1.966340e-06 -3.739640e-07   \n",
       "B-cells          0.000000e+00 -6.247138e-05  0.000000e+00  0.000000e+00   \n",
       "Nasal polyps     0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   \n",
       "Lymph node      -1.222434e-06  0.000000e+00  0.000000e+00 -2.092971e-05   \n",
       "Monocytes        0.000000e+00  3.257634e-05  0.000000e+00  4.918369e-07   \n",
       "Smooth muscle   -5.070917e-06  0.000000e+00  7.490819e-07 -1.284781e-06   \n",
       "Prostate         0.000000e+00  9.972075e-08  2.996328e-07  2.600362e-07   \n",
       "Colon            0.000000e+00 -1.283558e-06  1.449836e-06  8.528214e-07   \n",
       "Small intestine  0.000000e+00  4.639189e-07  0.000000e+00  4.834132e-08   \n",
       "Adrenal gland    0.000000e+00  0.000000e+00  0.000000e+00  1.117195e-06   \n",
       "NK-cells         0.000000e+00 -1.044446e-05  0.000000e+00  0.000000e+00   \n",
       "\n",
       "                 Q9Y6X9  Q9Y6Y0        Q9Y6Y8  \n",
       "tissue                                         \n",
       "Brain               0.0     0.0  0.000000e+00  \n",
       "Skeletal muscle     0.0     0.0  0.000000e+00  \n",
       "Tonsil              0.0     0.0  4.610759e-07  \n",
       "Liver               0.0     0.0  0.000000e+00  \n",
       "Parotid gland       0.0     0.0  0.000000e+00  \n",
       "Heart               0.0     0.0  1.866259e-07  \n",
       "Esophagus           0.0     0.0  4.898931e-07  \n",
       "Kidney              0.0     0.0  4.354605e-07  \n",
       "Testis              0.0     0.0 -2.924150e-07  \n",
       "Ovary               0.0     0.0  6.029453e-07  \n",
       "Placenta            0.0     0.0  0.000000e+00  \n",
       "Pituitary gland     0.0     0.0  0.000000e+00  \n",
       "Salivary gland      0.0     0.0  0.000000e+00  \n",
       "Lung                0.0     0.0  2.471847e-07  \n",
       "Urinary bladder     0.0     0.0  6.531908e-07  \n",
       "Thyroid             0.0     0.0  0.000000e+00  \n",
       "Adipose tissue      0.0     0.0  0.000000e+00  \n",
       "Pancreas            0.0     0.0  0.000000e+00  \n",
       "Appendix            0.0     0.0  3.919145e-07  \n",
       "Stomach             0.0     0.0 -3.418425e-07  \n",
       "Oviduct             0.0     0.0  0.000000e+00  \n",
       "Spleen              0.0     0.0  2.402550e-07  \n",
       "Duodenum            0.0     0.0  0.000000e+00  \n",
       "Endometrium         0.0     0.0 -7.097229e-06  \n",
       "Rectum              0.0     0.0  2.252390e-07  \n",
       "B-cells             0.0     0.0  3.603824e-07  \n",
       "Nasal polyps        0.0     0.0  0.000000e+00  \n",
       "Lymph node          0.0     0.0  1.321402e-06  \n",
       "Monocytes           0.0     0.0  0.000000e+00  \n",
       "Smooth muscle       0.0     0.0  8.033501e-07  \n",
       "Prostate            0.0     0.0  1.045105e-06  \n",
       "Colon               0.0     0.0  1.162524e-07  \n",
       "Small intestine     0.0     0.0  6.930432e-08  \n",
       "Adrenal gland       0.0     0.0  8.190510e-08  \n",
       "NK-cells            0.0     0.0  0.000000e+00  \n",
       "\n",
       "[35 rows x 4384 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test = MLMarker(baseline.iloc[0:1,:], binary = False, dev=False)\n",
    "prediction = MLMarker.adjusted_absent_shap_values_df(test, penalty_factor=1, n_preds=50)\n",
    "prediction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mlmarker",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
